### ISSUE TITLE
[BUG] `kak -l` never terminate

### ISSUE THREAD

**Issue by sucrecacao** [mawww/kakoune#4751]
### Version of Kakoune

v2020.09.01-1242-g24d60723

### Reproducer

Hard to tell, usually when I have multiple session running

### Outcome

running `kak -l` print a few sessions and  then does not terminate. Pressing `C-c` or `C-d` doesn't terminate the program neither

### Expectations

the program end and print the sessions ids on stdout

### Additional information

`kak -clear` doesn't terminate neither

**Comment by krobelus** [comment]
yeah I saw this occasionally as well but it's been a while.
Did you maybe SIGSTP the initial client (with Ctrl+Z)?

**Comment by sucrecacao** [comment]
might be... But I have like 7 workspace open, each runing at least 1000 terminals, and probably 1000000 kakoune session running everywhere ;) 


Is there a command that can help me found out ?

**Comment by krobelus** [comment]
Yes, each session is just a fifo socket the location depends on the first defined variable of:

	$XDG_RUNTIME_DIR/kakoune
	$TMPDIR/kakoune-$USER
	/tmp/kakoune-$USER

**Comment by sucrecacao** [comment]
Hmm weird, I only have a folder `kakoune-root` but I'm not root :(

**Comment by krobelus** [comment]
strange.. for me the fifos are in `$XDG_RUNTIME_DIR/kakoune` which is `/run/user/1000/kakoune`

**Comment by krobelus** [comment]
here's how you can find out:

```
$ strace -o kakoune.strace kak -s SESSION
$ grep SESSION kakoune.strace
execve("/home/johannes/bin/kak", ["kak", "-s", "SESSION"], 0x7ffc7873d900 /* 45 vars */) = 0
bind(3, {sa_family=AF_UNIX, sun_path="/run/user/1000/kakoune/SESSION"}, 110) = 0
unlink("/run/user/1000/kakoune/SESSION") = 0
```

**Comment by sucrecacao** [comment]
> yeah I saw this occasionally as well but it's been a while. Did you maybe SIGSTP the initial client (with Ctrl+Z)?

No I didn't

**Comment by sucrecacao** [comment]
My guess is that one of those 3 functions doesn't finish for some reason:
https://github.com/mawww/kakoune/blob/083bf82c235fa2821f279caa841c2765fbd81c68/src/file.cc#L439
https://github.com/mawww/kakoune/blob/b2c6bc46901368a1c30d47bd7718c2e10bc69480/src/remote.cc#L634
https://github.com/mawww/kakoune/blob/b2c6bc46901368a1c30d47bd7718c2e10bc69480/src/remote.cc#L590
